libnavigation-core/src/main/java/com/mapbox/navigation/core/NavigationFiles.kt

```diff
@@ -0,0 +1,55 @@
+package com.mapbox.navigation.core
+
+import android.content.Context
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.NavigationFiles.DirectoryType.CACHE
+import com.mapbox.navigation.core.NavigationFiles.DirectoryType.HISTORY
+import java.io.File
+
+internal class NavigationFiles(
+    val applicationContext: Context,
+    val logger: Logger
+) {
+
+    fun historyAbsolutePath(options: NavigationOptions): String =
+        absolutePath(options.historyRecorderOptions.fileDirectory, HISTORY)
+
+    fun cacheAbsolutePath(options: NavigationOptions): String =
+        absolutePath(options.routingTilesOptions.filePath, CACHE)
```

Comment:
I actually did something similar, and preferred to keep them separate. This is because it's unclear at this point where these features will live evolve. Can we follow the rule of 3 and do this later? https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
`HistoryFiles.absolutePath`
`RoutingTilesFiles.absolutePath`
had exactly the same logic with the only difference: default_path.

No it is handled in one place without code duplicating